AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift

```diff
@@ -91,14 +91,20 @@
         /// 2. check if the messageMonitor is allowing the message to be shown
         ///     a. if yes, show the message and exit the function
         ///     b. if no, call onShowFailure of the listener and exit the function
+        ///
+        /// - Parameter completion: called with `true` if message was shown
+        public func show(_ completion: @escaping ((Bool) -> Void)) {
+            show(withMessagingDelegateControl: true, completion)
+        }
+
         public func show() {
             show(withMessagingDelegateControl: true)
         }
 
-        public func show(withMessagingDelegateControl delegateControl: Bool) {
+        public func show(withMessagingDelegateControl delegateControl: Bool, _ completion: ((Bool) -> Void)? = nil) {
```

Comment:
public method isn't exposed to objc making this change safe for backwards compatibility via new parameter being optional